app/ai_sdk/agent/action.ts

```diff
@@ -11,37 +11,25 @@ export async function runAgent(input: string) {
   "use server";
 
   const stream = createStreamableValue();
-
   (async () => {
     const tools = [new TavilySearchResults({ maxResults: 1 })];
-
     const prompt = await pull<ChatPromptTemplate>(
       "hwchase17/openai-tools-agent",
     );
 
-    const llm = new ChatOpenAI({
-      model: "gpt-4o-mini",
-      temperature: 0,
-    });
+    const llm = new ChatOpenAI({ model: "gpt-4o-mini", temperature: 0 });
 
     const agent = createToolCallingAgent({
```

Comment:
Let's use `createReactAgent` and LangGraph here

Reply:
Will be done in a later PR